packages/core/src/symbols/symbol-flow.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  Context,
+  effect,
+  getContext,
+  onCleanup,
+} from "@alloy-js/core/jsx-runtime";
+import { isRef, Ref, shallowReactive } from "@vue/reactivity";
+import { MemberScopeContext, ScopeContext } from "../index.browser.js";
+import { OutputSymbol, OutputSymbolFlags } from "./output-symbol.js";
+
+export interface TakeSymbolCallback {
+  (symbol: OutputSymbol): void;
+}
+
+export interface TakeSymbolsCallback {
+  (symbols: Set<OutputSymbol>): void;
+}
+
+export function takeSymbols() {
+  const context = getContext()!;
+  context.takesSymbols = true;
+  context.takenSymbols = shallowReactive(new Set<OutputSymbol>());
+  return context.takenSymbols;
+}
+
+export function emitSymbol(
+  symbol: OutputSymbol | Ref<OutputSymbol | undefined>,
+) {
+  let ownerContext: Context | undefined;
+  let context = getContext()!.owner;
+  while (context) {
+    if (context.takesSymbols) {
+      ownerContext = context;
+      break;
+    }
+
+    if (
+      context.context &&
+      (context.context[ScopeContext.id] ||
+        context.context[MemberScopeContext.id])
+    ) {
+      // don't cross scope boundaries.
```

Comment:
If this is something we don't think will be hit normally, maybe a trace/warn would help pinpoint "unexpected" behavior related to this?

Reply:
No this will happen all the time when a symbol is emitted but nothing is around to take it. A trace seems appropriate tho.